runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -284,16 +284,50 @@ TR_VectorAPIExpansion::visitNodeToBuildVectorAliases(TR::Node *node)
       // check if VectorAPI method is supported
       TR::RecognizedMethod index = methodSymbol->getRecognizedMethod();
       int handlerIndex = index - _firstMethod;
-      //TODO: support mode per class
+
       if (methodElementType == TR::NoType ||
-          methodNumLanes == 0 ||
-          !methodTable[handlerIndex]._methodHandler(this, NULL, node, methodElementType, methodNumLanes, defaultCheckMode))
+          methodNumLanes == 0)
          {
          if (_trace)
-            traceMsg(comp(), "Invalidating #%d due to unsupported elementType=%d, numLanes=%d, or opcode in node %p\n",
+            traceMsg(comp(), "Invalidating #%d due to unknown elementType=%d, numLanes=%d in node %p\n",
                      node->getSymbolReference()->getReferenceNumber(), (int)methodElementType, methodNumLanes, node);
          invalidateSymRef(node->getSymbolReference());
          }
+      else
+         {
+         int32_t elementSize = OMR::DataType::getSize(methodElementType);
+         vec_sz_t vectorLength = methodNumLanes*8*elementSize;
+
+         bool canVectorize = methodTable[handlerIndex]._methodHandler(this, NULL, node, methodElementType, vectorLength,
+                                                                      checkVectorization);
+         bool canScalarize = methodTable[handlerIndex]._methodHandler(this, NULL, node, methodElementType, vectorLength,
+                                                                         checkScalarization);
+         if (!canVectorize)
+            {
+            if (_trace)
+               traceMsg(comp(), "Can't vectorize #%d due to unsupported opcode in node %p\n",
+                                 node->getSymbolReference()->getReferenceNumber(), node);
+
+            _aliasTable[methodRefNum]._cantVectorize = true;
+
+            if (!canScalarize)
+               {
+               _aliasTable[methodRefNum]._cantScalarize = true;
+               if (_trace)
+                  traceMsg(comp(), "Invalidating #%d due to unsupported opcode in node %p\n",
+                           node->getSymbolReference()->getReferenceNumber(), node);
+               invalidateSymRef(node->getSymbolReference());
```

Comment:
Just confirming we are invalidating the symref on this path because we can't vectorize nor scalarize. This is why this call is missing on the `else if` path, right?

Reply:
yes, since we can't do either we just invalidate.